### ISSUE TITLE
Deselecting indexable Multiselect product attributes causes a deprecation warning

### ISSUE THREAD

**Issue by ghost** [magento/magento2#36530]
### Preconditions and environment

- Magento version: 2.4.4-p1, 
- PHP 8.1
- Developer mode
- Having an **indexable** (i.e. having one of `is_filterable_in_search`, `is_visible_in_advanced_search` or `is_filterable` set) `multiselect` product attribute

### Steps to reproduce

1. Select one or more options for the multiselect
2. Save the product
3. Deselect all options
4. Save the product

### Expected result

The corresponding row in `catalog_product_attribute_text` is deleted, no warnings thrown.

### Actual result

The page informs it was saved correctly, but:
1. The following critical notice is logged:
```
[2022-11-24T18:05:11.144131+00:00] main.CRITICAL: Exception: Deprecated Functionality: explode(): Passing null to parameter #2 ($string) of type string is deprecated in vendor/magento/module-catalog/Model/ResourceModel/Product/Indexer/Eav/Source.php on line 445 in vendor/magento/framework/App/ErrorHandler.php:61
```
2. The `catalog_product_attribute_text` table keeps a `NULL` value for that attribute.
3. An `indexer:reindex` throws the same exception for that attribute:
```
Product EAV index process error during indexation process:
Deprecated Functionality: explode(): Passing null to parameter #2 ($string) of type string is deprecated in vendor/magento/module-catalog/Model/ResourceModel/Product/Indexer/Eav/Source.php on line 445 in vendor/magento/framework/App/ErrorHandler.php:61
```

Furthermore, saving the product again with no changes removes the row from `catalog_product_attribute_text` and allows the reindex to finish.

### Additional information

I have not tested this on production mode.
For the moment I'll be patching this line via `composer` patch, but I'm not sure whether having the `NULL` row affects any other functionality.
```php
$values = $row['value'] ? explode(',', $row['value']) : [];
```

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
Hi @francisco-serfe,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.


**Steps to reproduce**

1.Select one or more options for the multiselect
2.Save the product
3.Deselect all options
4.Save the product


Product saved successfully.No warnings thrown.

![vhjasdjh-Products-Inventory-Catalog-Magento-Admin](https://user-images.githubusercontent.com/51680745/203967351-85bc380e-8b18-4992-b2bb-87ef4eadd11c.png)




![vhjasdjh-Products-Inventory-Catalog-Magento-Admin (1)](https://user-images.githubusercontent.com/51680745/203967385-4dc0dfc3-b5ee-4894-9830-85403ed78efa.png)


No error for `indexer:reindex`



<img width="650" alt="Screenshot 2022-11-25 at 1 52 52 PM" src="https://user-images.githubusercontent.com/51680745/203967619-b565c836-8e00-4893-b422-1c33763dbe78.png">



Kindly recheck the behavior on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.


Thanks.

**Comment by francisco-serfe** [comment]
Sorry, I missed a precondition for the attribute. It should be indexable, per [_getIndexableAttributesCondition](https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Eav/AbstractEav.php#L250-L261):
```php
    protected function _getIndexableAttributesCondition()
    {
        $conditions = [
            'ca.is_filterable_in_search > 0',
            'ca.is_visible_in_advanced_search > 0',
            'ca.is_filterable > 0',
            // Visibility is attribute that isn't used by search, but required to determine is product should be shown
            "ea.attribute_code = 'visibility'"
        ];


        return implode(' OR ', $conditions);
    }
```

In my specific case, the attribute has `is_filterable` enabled.

**Comment by engcom-Bravo** [comment]
November 25,2022

**Comment by engcom-Bravo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
Hi @francisco-serfe,

Thanks for your update.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.



**Steps to reproduce**

1.Select one or more options for the multiselect
2.Save the product
3.Deselect all options
4.Save the product


<img width="1408" alt="Screenshot 2022-11-28 at 2 32 04 PM" src="https://user-images.githubusercontent.com/51680745/204269810-276767d7-a533-47e0-a0c1-8156c18bf819.png">




<img width="1356" alt="Screenshot 2022-11-28 at 2 32 26 PM" src="https://user-images.githubusercontent.com/51680745/204269851-a255bde0-15fb-4245-8b45-95e360190837.png">




![vhjasdjh-Products-Inventory-Catalog-Magento-Admin (2)](https://user-images.githubusercontent.com/51680745/204269953-0fde9cb4-927c-4d00-9f62-887bb32206dd.png)




![vhjasdjh-Products-Inventory-Catalog-Magento-Admin (3)](https://user-images.githubusercontent.com/51680745/204270002-7a107f80-dc09-44da-ada0-4c04541bdabc.png)



No error for `indexer:reindex`

<img width="643" alt="Screenshot 2022-11-28 at 2 31 19 PM" src="https://user-images.githubusercontent.com/51680745/204270317-68d4b571-44f5-41ef-ba99-ddccdd18092e.png">


Kindly recheck the behavior on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.


Thanks.

**Comment by francisco-serfe** [comment]
Well, I specifically mentioned I was reproducing this behavior in `2.4.4-p1`. Looks like it was fixed in `2.4.5` via [this commit](https://github.com/magento/magento2/commit/d255ba3fb31db0bea0f8b9fea96d33a85cf5e091). Feel free to close.

**Comment by engcom-Bravo** [comment]
Hi @francisco-serfe,

Thanks for your Update.

We are Closing this issue as per your comment as issue is not reproducible and issue got fixed.

Thanks.
